midterm-2023-11-07/code/5040-dara-poposka/script.ts

```diff
@@ -0,0 +1,116 @@
+interface Movie{
+    id: number;
+    title: string;
+    year: number;
+    director: string;
+    genre: string[];
+    plot: string;
+    cast: {
+      actor: string;
+      character: string;
+    }[];
+    oscars: { [oscarType: string]: string };
+    rating: number;
+  }
+  type MovieSorter = (first: Movie, second: Movie) => number;
+  
+  document.addEventListener("DOMContentLoaded", siteCode);
+  
+  let movies: Movie[] = [];
+  
+  async function siteCode() {
+    const data = await loadData();
+    movies = data;
+  
+    displayMovies(movies);
+  
+    const titleSort = document.getElementById("sort-title")!;
```

Comment:
Adding handlers for the sort and filter elements +5
Using a `change` event extra +5